@navikt/core/react/src/form/combobox/FilteredOptions/useVirtualFocus.ts

```diff
@@ -89,6 +86,32 @@ const useVirtualFocus = (
     }
   };
 
+  const moveFocusUpBy = (numberOfElements: number) => {
+    const elementsAbleToReceiveFocus = getElementsAbleToReceiveFocus();
+    if (!activeElement) {
+      return;
+    }
+    const currentIndex = elementsAbleToReceiveFocus.indexOf(activeElement);
+    const newIndex = Math.max(currentIndex - numberOfElements, 0);
+    setActiveElement(elementsAbleToReceiveFocus[newIndex]);
+  };
+
+  const moveFocusDownBy = (numberOfElements: number) => {
+    const elementsAbleToReceiveFocus = getElementsAbleToReceiveFocus();
+    const currentIndex = activeElement
+      ? elementsAbleToReceiveFocus.indexOf(activeElement)
+      : -1;
+    const newIndex = Math.min(
+      currentIndex + numberOfElements,
+      elementsAbleToReceiveFocus.length - 1,
+    );
+    setActiveElement(elementsAbleToReceiveFocus[newIndex]);
+  };
+
+  useLayoutEffect(() => {
+    activeElement?.scrollIntoView?.({ block: "nearest" });
+  }, [activeElement]);
+
```

Comment:
Is there a reason `_moveFocusAndScrollTo` were scrapped and Effect was used instead?

Reply:
In some cases while testing, I noticed we did not scroll as we should have. My guess it was a race condition, where the element either unmounted and remounted, or something like that.

Moving the scrolling out of `_moveFocusAndScrollTo` made sure we always scroll to the element when the index is changed, and after doing that, the function no longer made sense, as it just wrapped `setActiveElement`.

I am open for re-testing to see if we could have fixed this inside of `_moveFocusAndScrollTo`.

Also, if we keep this implementation, is `useLayoutEffect` correct? My thought behind it was that scrolling should be done while/before moving focus, but now I am second guessing myself.

Reply:
Since an actual HtmlElement element is passed into `setActiveElement` and `_moveFocusAndScrollTo` before, scroll should in theory always work, but i might be missing some details here. 
As for `useLayoutEffect`, i think `useEffect` would achieve the same experience, as long as the visuals for the "active" element and the scroll is in sync.

Reply:
Yeah, that does sound right. I'll give it another go, and see if I can re-create the bug or if it may have been a temporary glitch.

Reply:
Was it a specific story so i can try to replicate it?

Reply:
Just the default one. I don't remember the exact order of keystrokes I used, so I will have to just test randomly.

Reply:
Try the following:

1. Focus the input
2. Press End (move focus to bottom of the list)
3. Press Home (focus the input and dropdown will close)
4. Press Arrow Down (focus moves to first element in list, but the list is scrolled all the way down 🐛 )
5. Press Arrow Down again (focus moves to second element and is scrolled into view)

Reply:
Makes sense since the element isn't rendered yet then 👍 Can't see any noticeable difference between layoutEffect and useEffect when testing, but if we use layoutEffect we have to use `useClientLayoutEffect` from `/utils` to support SSR like nextjs.

Reply:
useEffect seems to work fine. I'll change it to use that instead.